zerver/tests/test_zcommand.py

```diff
@@ -81,3 +83,67 @@ def test_fixed_zcommand(self) -> None:
         result = self.client_post("/json/zcommand", payload)
         self.assert_json_success(result)
         self.assert_in_response('You are still in fixed width mode', result)
+
+    def test_digress_zcommand(self) -> None:
+        self.login('hamlet')
+        hamlet = self.example_user('hamlet')
+        hamlet.save()
+
+        self.make_stream('general')
+        self.subscribe(hamlet, 'general')
+        self.make_stream('test')
+        self.subscribe(hamlet, 'test')
+
+        payload = dict(
+            command="/digress",
+            command_data=orjson.dumps(dict(
+                old_stream="general",
+            )).decode(),
+        )
+        result = self.client_post("/json/zcommand", payload)
+        self.assert_json_error(result, "Invalid data.")
+
+        payload = dict(
+            command="/digress",
+            command_data= orjson.dumps(dict(
+                old_stream="general",
+                old_topic="old topic name",
+                new_stream="test",
+                new_topic="new topic name",
+            )).decode(),
+        )
+        result = self.client_post("/json/zcommand", payload)
+        self.assert_json_success(result)
+        self.assertIn("Old topic #**general>old topic name** does not exist.", result.json()['msg'])
+
+        self.client_post("/json/messages", {"type": "stream",
+                                            "to": "general",
+                                            "client": "test suite",
+                                            "content": "Test message",
+                                            "topic": "old topic name"})
+
+        payload = dict(
+            command="/digress",
+            command_data=orjson.dumps(dict(
+                old_stream="general",
+                old_topic="old topic name",
+                new_stream="general",
+                new_topic="old topic name",
+            )).decode(),
+        )
+        result = self.client_post("/json/zcommand", payload)
+        self.assert_json_success(result)
+        self.assertIn("Cannot digress to the same thread.", result.json()['msg'])
+
+        payload = dict(
+            command="/digress",
+            command_data=orjson.dumps(dict(
+                old_stream="general",
+                old_topic="old topic name",
+                new_stream="general",
+                new_topic="new topic name",
+            )).decode(),
+        )
+        result = self.client_post("/json/zcommand", payload)
+        self.assert_json_success(result)
+        self.assertEqual('', result.json()['msg'])
```

Comment:
You should verify what was the message that was sent in the both the threads here.